test/unit/mallctl.c

```diff
@@ -711,6 +711,53 @@ TEST_BEGIN(test_arena_i_dss) {
 }
 TEST_END
 
+TEST_BEGIN(test_arena_i_name) {
+	unsigned arena_ind;
+	size_t ind_sz = sizeof(arena_ind);
+	const char *name_new;
+	char name_old[ARENA_NAME_LEN];
+	char *name_oldp = name_old;
+	size_t sz = sizeof(name_oldp);
+	size_t mib[3];
+	size_t miblen;
+	char super_long_name[] = "A name longer than ARENA_NAME_LEN";
+	int super_long_name_len = 33;
+
+	miblen = sizeof(mib)/sizeof(size_t);
+	expect_d_eq(mallctlnametomib("arena.0.name", mib, &miblen), 0,
+	    "Unexpected mallctlnametomib() error");
+
+	expect_d_eq(mallctl("arenas.create", (void *)&arena_ind, &ind_sz, NULL, 0),
+	    0, "Unexpected mallctl() failure");
+	mib[1] = arena_ind;
+
+	expect_d_eq(mallctlbymib(mib, miblen, (void *)&name_oldp, &sz,
+	    NULL, 0), 0, "Unexpected mallctl() failure");
+	expect_str_eq(name_old, "manual", "Unexpected default value for arena name");
+
+	name_new = "test name";
+	expect_d_eq(mallctlbymib(mib, miblen, NULL, NULL, (void *)&name_new,
+	    sizeof(name_new)), 0, "Unexpected mallctl() failure");
+
+	expect_d_eq(mallctlbymib(mib, miblen, (void *)&name_oldp, &sz,
+	    NULL, 0), 0, "Unexpected mallctl() failure");
+	expect_str_eq(name_old, "test name", "Unexpected value for arena name");
```

Comment:
nit: use name_new instead of "test name" to avoid having the same string twice.

Reply:
modified